### ISSUE TITLE
Subdocument Hooks not triggered in Discriminators

### ISSUE THREAD

**Issue by kticka** [Automattic/mongoose#15092]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have searched existing issues to ensure the bug has not already been reported

### Mongoose version

8.8.4

### Node.js version

22.12.0

### MongoDB server version

8.0.3

### Description

Hi, it seems, that hooks of embedded documents aren't triggered properly when used with discriminators. I wrote very simple example to reproduce.

### Steps to Reproduce

Document.js
```
const mongoose    = require('mongoose')
const Subdocument = require('./Subdocument')

const schema = mongoose.Schema({
  name:         String,
  subdocuments: [Subdocument]
}, {discriminatorKey: 'type'})

schema.pre('save', function (next) {
  console.log(`Saving Document: ${this.name}`)
  next()
})

module.exports = mongoose.model('Document', schema)
```

Subdocument.js:
```
const mongoose = require('mongoose')

const schema = mongoose.Schema({
  name: String
})

schema.pre('save', function(next) {
  console.log(`Saving Subdocument: ${this.name}`)
  next()
})

module.exports = schema
```

Discriminator.js:
```
const mongoose = require('mongoose')
const Document = require('./Document')

const schema = mongoose.Schema({})

schema.pre('save', function(next) {
  console.log(`Saving Discriminator: ${this.name}`)
  next()
})

module.exports = Document.discriminator('Discriminator', schema)
```

Test.js
```
module.exports      = async function () {
  const document     = new Document({name: 'Document'})
  const dicriminator = new Discriminator({name: 'Discriminator', type: 'Discriminator'})

  document.subdocuments.push({name: 'Document Subdocument'})
  dicriminator.subdocuments.push({name: 'Discriminator Subdocument'})

  await document.save()
  await dicriminator.save()

  return 'Done'
}
```

### Expected Behavior

When saving either Document or Discriminator, the Subdocument hook should be triggered. The expected output is:
```
Saving Subdocument: Document Subdocument
Saving Document: Document
Saving Subdocument: Discriminator Subdocument
Saving Document: Discriminator
Saving Discriminator: Discriminator
```

However, when saving the Discriminator, the Subdocument hook is not triggered. The actual output is:
```
Saving Subdocument: Document Subdocument
Saving Document: Document
Saving Document: Discriminator
Saving Discriminator: Discriminator
```

**Comment by kticka** [comment]
After further testing, I found that the issue does not occur in Mongoose 8.8.2. The problem starts in version 8.8.3 and persists in 8.8.4.

**Comment by maximilianschmid** [comment]
The problem starts with version 8.8.3 and is still there in 8.8.4 and 8.9. In my case the supdocuments with discriminators don't get the modified date updated.

**Comment by vkarpov15** [comment]
Confirmed that this issue will be fixed by #15099.
